### PR TITLE
Adds Proxy Protocol Support

### PR DESCRIPTION
**Pull request by danehans**
Adds support for managing the [proxy protocol](https://kubernetes.io/docs/concepts/services-networking/service/#proxy-protocol-support-on-aws) for AWS Classic load balancers.

Requires: https://github.com/projectcontour/contour-operator/pull/366

Fixes: https://github.com/projectcontour/contour-operator/issues/49

xref: https://projectcontour.io/guides/proxy-proto/

### CODE REVIEW

**Comment by danehans** [comment]
Since this PR requires https://github.com/projectcontour/contour-operator/pull/366, filter your review to [this commit](https://github.com/projectcontour/contour-operator/pull/367/commits/8dbf786837eae6414e4dc7fe17d1b53af929af85).

**Comment by youngnick** [comment]
I've just merged #366, this needs a rebase now @danehans.

**Review by youngnick — APPROVED** [review]

File: internal/objects/service/service.go [hunk 1]
```diff
@@ -240,13 +244,16 @@ func DesiredEnvoyService(contour *operatorv1alpha1.Contour) *corev1.Service {
...
+	// Add AWS LB annotations based on the network publishing strategy and provider type.
+	if contour.Spec.NetworkPublishing.Envoy.Type == operatorv1alpha1.LoadBalancerServicePublishingType {
+		// Add the TCP backend protocol annotation for AWS classic load balancers.
+		if isELB(&contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters) {
+			svc.Annotations[awsLbBackendProtoAnnotation] = "tcp"
+			svc.Annotations[awsLBProxyProtocolAnnotation] = "*"
+		} else {
+			// Annotate the service for an NLB.
+			svc.Annotations[awsLBTypeAnnotation] = "nlb"
+		}
```
Comment (skriss) [inline internal/objects/service/service.go:256]: should we be checking for `params.Type == operatorv1alpha1.AWSLoadBalancerProvider` here to only add the annotation if AWS is requested? It looks like the previous code did that. Worth adding some tests for too.

**Review by skriss — CHANGES_REQUESTED** [review]

**Review by AskAlice — APPROVED** [review]

File: internal/objects/service/service.go (hunk 1 above)
Comment (danehans) [reply internal/objects/service/service.go:257]: @skriss thanks for catching this. Commit `bbfec50` fixes this issue, PTAL.

**Review by skriss — APPROVED** [review]
LGTM. I have a refactor in mind for the test code to check each service for an exact set of annotations, but I'm happy to submit that in a follow-up.

**Pull request merged**